flow/record/fieldtypes/__init__.py

```diff
@@ -280,23 +277,47 @@ def __new__(cls, *args, **kwargs):
             if isinstance(arg, bytes_type):
                 arg = arg.decode("utf-8")
             if isinstance(arg, string_type):
-                # I expect ISO 8601 format e.g. datetime.isoformat()
-                # String constructor is used for example in JsonRecordAdapter
-                # Note: ISO 8601 is fully implemented in fromisoformat() from Python 3.11 and onwards.
-                # Until then, we need to manually detect timezone info and handle it.
-                if not PY_311 and any(z in arg[19:] for z in ["Z", "+", "-"]):
-                    spec = ISO_FORMAT_WITH_MS if "." in arg[19:] else ISO_FORMAT
-                    try:
-                        obj = cls.strptime(arg, spec)
-                    except ValueError:
-                        # Sometimes nanoseconds need to be stripped
-                        obj = cls.strptime(re.sub(RE_STRIP_NANOSECS, "\\1", arg), spec)
-                else:
-                    try:
-                        obj = cls.fromisoformat(arg)
-                    except ValueError:
-                        # Sometimes nanoseconds need to be stripped
-                        obj = cls.fromisoformat(re.sub(RE_STRIP_NANOSECS, "\\1", arg))
+                # If we are on Python 3.11 or newer, we can use fromisoformat() to parse the string (fast path)
+                #
+                # Else we need to do some manual parsing to fix some issues with the string format:
+                # - Python 3.10 and older do not support nanoseconds in fromisoformat()
+                # - Python 3.10 and older do not support Z as timezone info in fromisoformat()
+                # - Python 3.10 and older do not support +0200 as timezone info in fromisoformat()
+                # - Python 3.10 and older requires "T" between date and time in fromisoformat()
+                #
+                # There are other incompatibilities, but we don't care about those for now.
+                if not PY_311:
+                    # Convert Z to +00:00 so that fromisoformat() works correctly on Python 3.10 and older
+                    if arg.endswith("Z"):
```

Comment:
```suggestion
                    if arg[-1] == "Z":
```
A rudimentary `%timeit` suggests that this is faster.